ext/opcache/Optimizer/block_pass.c

```diff
@@ -74,6 +74,17 @@ int zend_optimizer_get_persistent_constant(zend_string *name, zval *result, int
 
 /* Data dependencies macros */
 
+static zend_always_inline uint32_t linear_var_num(zend_op_array *op_array, uint32_t var) {
+	uint32_t num = VAR_NUM(var);
+	if (0 > (int32_t) num) {
+		return - 1 - num;
+	} else {
+		return num + op_array->num_args;
+	}
+}
+#undef VAR_NUM
+#define VAR_NUM(var) linear_var_num(op_array, var)
+
```

Comment:
Changed meaning of op_array->last_var and LOCALS/ARGUMENTS numbering affects optimization framework.